### PR TITLE
Issue #7 Adapt Ehcache 3

### PR DESCRIPTION
**Pull request by ljacomet**
- Remove from profile and move to top level module
- Use released 3.0.1
- Adapt TCK configuration

### CODE REVIEW

**Comment by ljacomet** [comment]
@cruftex Note that this does not take care of adding the current excludes from Ehcache 3.
I could not figure out the way of doing that.

File: ehcache-V3-test/pom.xml [hunk 1]
```diff
@@ -20,17 +20,17 @@
...
+    <CacheManagerImpl>org.ehcache.CacheManager</CacheManagerImpl>
+    <CacheImpl>org.ehcache.Cache</CacheImpl>
+    <CacheEntryImpl>org.ehcache.Cache$Entry</CacheEntryImpl>
 
     <!--The MBeanServer, so that the TCK can resolve it. -->
-    <javax.management.builder.initial>org.ehcache.jsr107.Eh107MBeanServerBuilder
+    <javax.management.builder.initial>org.ehcache.jsr107.internal.tck.Eh107MBeanServerBuilder
     </javax.management.builder.initial>
-    <org.jsr107.tck.management.agentId>anyvalue</org.jsr107.tck.management.agentId>
+    <org.jsr107.tck.management.agentId>Eh107MBeanServer</org.jsr107.tck.management.agentId>
```
Comment (cruftex) [inline ehcache-V3-test/pom.xml]: Please try with `anyvalue`, it works unless you changed something. I use `anyvalue` to say that the value of the parameter isn't relevant, it just needs to be something.

Or do you have applications that use the Eh107MBeanServer agant id, too?

BTW, do you have an opinion on:
https://github.com/jsr107/jsr107tck/issues/83

**Comment by cruftex** [comment]
> Note that this does not take care of adding the current excludes from Ehcache 3.
> I could not figure out the way of doing that.

AFIAK needs to go into `src/main/resources`.

File: ehcache-V3-test/pom.xml (hunk 1 above)
Comment (ljacomet) [inline ehcache-V3-test/pom.xml]: Revert the change to keep `anyvalue`

Will look at the linked issue.

**Comment by ljacomet** [comment]
From what I see, the changes have to go in the root `ExcludeList` file, which means it gets applied to all implementations.
I could not find a way to have these defined per implementation.

**Comment by cruftex** [comment]
Can you please try: `ehcache-V3-test/src/main/resources`

**Comment by ljacomet** [comment]
My bad, had tried `ehcache-V3-test/src/test/resources` however the other location does not help either.
The only place where it seems picked up properly is when put in the existing `ExcludeList` of your project, at the root.

**Comment by cruftex** [comment]
Sorry, and thanks for trying!

I can see where the problem is. Needs some reorganizing.

**Comment by ljacomet** [comment]
From what I can tell, none of the other implementations explicitly exclude tests, so if ok with you this could be merged as-is.
In any case the 1.1 release of the TCK includes the fixes that would enable use to stop skipping the two real tests that are excluded.

**Comment by cruftex** [comment]
Should be fixed now! 

Pls. git pull
Test is the correct place: `ehcache-V3-test/src/test/resources`

**Comment by ljacomet** [comment]
Updated and rebased PR.

Though I had to re-order your resources declaration to make it work.

**Comment by cruftex** [comment]
Formidable!
